cms/djangoapps/contentstore/rest_api/v0/tests/test_tabs.py

```diff
@@ -0,0 +1,225 @@
+
+        # remove the middle tab
+        # (the code needs to handle the case where tabs requested for re-ordering is a subset of the tabs in the course)
+        removed_tab = tab_ids.pop(num_orig_tabs // 2)
+        assert len(tab_ids) == num_orig_tabs - 1
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in tab_ids]},
+            content_type="application/json",
+        )
+        assert resp.status_code == 204
+
+        # reload the course and verify the new tab order
+        self.reload_course()
+        new_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        assert new_tab_ids == tab_ids + [removed_tab]
+        assert new_tab_ids != orig_tab_ids
+
+    def test_reorder_tabs_invalid_list(self):
+        """Test re-ordering of tabs with invalid tab list"""
+
+        orig_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        tab_ids = list(orig_tab_ids)
+
+        # reorder the first two tabs
+        tab_ids[0], tab_ids[1] = tab_ids[1], tab_ids[0]
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in tab_ids]},
+            content_type="application/json",
+        )
+        assert resp.status_code == 400
+        error = self.check_invalid_response(resp)
+        assert "error" in error
+
+    def test_reorder_tabs_invalid_tab(self):
+        """Test re-ordering of tabs with invalid tab"""
+
+        invalid_tab_ids = ["courseware", "info", "invalid_tab_id"]
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in invalid_tab_ids]},
+            content_type="application/json",
+        )
+        self.check_invalid_response(resp)
+
+    def check_toggle_tab_visibility(self, tab_type, new_is_hidden_setting):
+        """Helper method to check changes in tab visibility"""
+        old_tab = CourseTabList.get_tab_by_type(self.course.tabs, tab_type)
+        # visibility should be different from new setting
+        assert old_tab.is_hidden != new_is_hidden_setting
+
+        resp = self.client.post(
+            self.url,
+            data=json.dumps(
+                {
+                    "tab_id_locator": {"tab_id": old_tab.tab_id},
+                    "is_hidden": new_is_hidden_setting,
+                }
+            ),
+            content_type="application/json",
+        )
+        assert resp.status_code == 204
+        # reload the course and verify the new visibility setting
+        self.reload_course()
+        new_tab = CourseTabList.get_tab_by_type(self.course.tabs, tab_type)
+        assert new_tab.is_hidden == new_is_hidden_setting
+
+    def test_toggle_tab_visibility(self):
+        """Test that toggling the visibility via the API works."""
+        self.check_toggle_tab_visibility("wiki", True)
```

Comment:
maybe => `update_and_check_tab_visibility`

Reply:
The current name specifies that it is toggling visibility, `update` is less specific.